lib/capn.c

```diff
@@ -7,6 +7,8 @@
  * of the MIT license.  See the LICENSE file for details.
  */
 
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
```

Comment:
The previous PR fixed two cases in the generated code. There are extra cases here in the runtime  library.

I considered fixing each of the cases, but the relevant struct (`capn_ptr`) has 9 fields and so it seemed somewhat verbose and brittle to do so.

Happy to use `#ifdef __GNUC__`